libtenzir/builtins/functions/community_id.cpp

```diff
@@ -0,0 +1,140 @@
+          .primary(args.src_ip)
+          .emit(ctx);
+        return null_series();
+      }
+      auto dst_ip_series = eval(args.dst_ip);
+      if (not caf::holds_alternative<ip_type>(dst_ip_series.type)) {
+        diagnostic::warning("`community_id` expected `ip` as 3rd argument")
+          .primary(args.dst_ip)
+          .emit(ctx);
+        return null_series();
+      }
+      auto src_port_series = eval(args.src_port);
+      if (not caf::holds_alternative<int64_type>(src_port_series.type)) {
+        diagnostic::warning("`community_id` expected `integer` as 2nd argument")
+          .primary(args.src_port)
+          .emit(ctx);
+        return null_series();
+      }
+      auto dst_port_series = eval(args.dst_port);
+      if (not caf::holds_alternative<int64_type>(dst_port_series.type)) {
+        diagnostic::warning("`community_id` expected `integer` as 4th argument")
+          .primary(args.dst_port)
+          .emit(ctx);
+        return null_series();
+      }
+      auto proto_series = eval(args.proto);
+      if (not caf::holds_alternative<string_type>(proto_series.type)) {
+        diagnostic::warning("`community_id` expected `string` as 5th argument")
+          .primary(args.dst_port)
+          .emit(ctx);
+        return null_series();
+      }
+      auto b = arrow::StringBuilder{};
+      check(b.Reserve(eval.length()));
+      auto src_ips = src_ip_series.as<ip_type>();
+      auto dst_ips = dst_ip_series.as<ip_type>();
+      auto src_ports = src_port_series.as<int64_type>();
+      auto dst_ports = dst_port_series.as<int64_type>();
+      auto protos = proto_series.as<string_type>();
+      for (auto i = int64_t{0}; i < eval.length(); ++i) {
+        if (src_ips->array->IsNull(i) or src_ports->array->IsNull(i)
+            or dst_ips->array->IsNull(i) or dst_ports->array->IsNull(i)
+            or protos->array->IsNull(i)) {
+          check(b.AppendNull());
+          continue;
+        }
+        const auto* src_ip_ptr = src_ips->array->storage()->GetValue(i);
+        const auto* dst_ip_ptr = dst_ips->array->storage()->GetValue(i);
+        auto src_ip = ip::v6(as_bytes<16>(src_ip_ptr));
+        auto dst_ip = ip::v6(as_bytes<16>(dst_ip_ptr));
+        auto src_port = src_ports->array->GetView(i);
+        auto dst_port = dst_ports->array->GetView(i);
+        auto proto = protos->array->GetView(i);
+        auto type = port_type::unknown;
+        if (proto == "tcp") {
+          type = port_type::tcp;
+        } else if (proto == "udp") {
+          type = port_type::udp;
+        } else if (proto == "icmp") {
+          type = port_type::icmp;
+        } else if (proto == "icmp6") {
+          type = port_type::icmp6;
+        } else {
+          diagnostic::warning("`community_id` expected `tcp`, `udp`, `icmp`, "
+                              "or `icmp6` as protocol")
+            .primary(args.proto)
+            .emit(ctx);
+          return null_series();
```

Comment:
I don't think you want to return a whole batch of nulls if one protocol in the batch is invalid, but rather just add a null here and emit a warning.

Reply:
Yes, thanks. Fixed.